vm/src/vm/runners/cairo_runner.rs

```diff
@@ -1194,6 +1194,16 @@ impl MulAssign<usize> for ExecutionResources {
 
 #[cfg(feature = "parity-scale-codec")]
 impl Encode for ExecutionResources {
+    fn size_hint(&self) -> usize {
+        8 + 8
```

Comment:
n_steps = `size_of::<u64>()`
n_memory_holes = `size_of::<u64>()`
len of instance counters = `size_of::<u8>()` (there at most is one entry by builtin. Most likely there won't be more than 31 builtins. Currently there is 9 of them) (the len is encoded using it's compact form, therefore it fit on a u8)

For the last part, I don't love at all the idea of iterating over the hashmap a first time there but I don't think we have a choice.